### DISCUSSION TITLE
Item stack at the core level

### DISCUSSION THREAD

**Discussion by groundbreakingmc** [PaperMC/Paper#11070] [Ideas]
Hey, how about adding a feature to change the max stack size for items at the core level? Third-party plugins often struggle with this, whether they’re using NMS or BukkitEvents. Plus, those plugins usually don’t support creative mode! Patching the core to add this feature isn’t something everyone wants to do, especially since it requires more than beginner programming skills.

https://github.com/user-attachments/assets/ea9f17f5-d217-4eac-9774-dfaf98a7d620

**Comment by masmc05** [comment]
https://jd.papermc.io/paper/1.21/org/bukkit/inventory/meta/ItemMeta.html#setMaxStackSize(java.lang.Integer)

you can already do that

**Reply by groundbreakingmc** [reply]
From which version would this work? 🧐

**Reply by masmc05** [reply]
1.20.5

**Reply by groundbreakingmc** [reply]
> 1.20.5

😔 ok, thx

**Reply by lynxplay** [reply]
Closing this as, yea, on a per itemstack basis this is already possible and, until the Item registry is data driven, it won't be nicely doable with client support for the whole Item type.
